include/kwk/utility/traits/as_dimension.hpp

```diff
@@ -68,9 +68,9 @@ namespace kwk
 
 namespace kwk::__
 {
-  KWK_TRIVIAL constexpr auto as_dimension_(KWK_DELAY(), joker               , std::integral auto d) { return d; }
-  KWK_TRIVIAL constexpr auto as_dimension_(KWK_DELAY(), concepts::axis auto , std::integral auto d) { return d; }
-  KWK_TRIVIAL constexpr auto as_dimension_(KWK_DELAY(), std::integral auto v, std::integral auto  ) { return v; }
+  KWK_TRIVIAL constexpr auto as_dimension_(KWK_DELAY(), joker                , std::integral auto d) { return d; }
+  KWK_TRIVIAL constexpr auto as_dimension_(KWK_DELAY(), concepts::axis auto v, std::integral auto  ) { return v.value; }
```

Comment:
why ?

Reply:
well it was just wrong/incorrect as it was? as_dimension is supposed to return the 'extent' of 'v' is it not?
my code didn't work properly until i changed that

(why is as_dimension implemented with all this extra machinery is completely different question :D)

Reply:
purely from looking at the code/'logicwise': an axis has its value - why would you return some 'default' (d) instead of that (just like you return the 'left'/v value for integral and constant)

Reply:
sometimes you don't have a value. You can have a shape<width,height> those are axis but has no value.

Reply:
axis only has value when used in constexpr context like `shape<my_axis[4]>`.
At runtime, their value is passed via = as in `shape x{ my_axis = 4}`

Now this is something we may need to polish.